mesonbuild/build.py

```diff
@@ -2257,6 +2263,12 @@ def get_link_dep_subdirs(self):
     def get_all_link_deps(self):
         return []
 
+    def is_internal(self) -> bool:
+        return False
```

Comment:
`is_internal()` is defined on `BuildTarget` as:

```py
def is_internal(self):
    return isinstance(self, StaticLibrary) and not self.need_install
```

I think in this case we can do something like:

```py
def is_internal(self) -> bool:
    if not self.should_install():
        return True
    for out in self.get_outputs():
        # Can't check if this is a static library, so try to guess
        if not out.endswith(('.a', '.lib')):
            return False
    return True
```